### ISSUE TITLE
export * as ns does not check for missing modules in System bundles

### ISSUE THREAD

**Issue by grosto** [microsoft/TypeScript#38227]
**TypeScript Version:**  3.8.3

**Search Terms:** System, SystemJS, Bundle, export * as n

**Expected behavior:**
Should throw an error and do not bundle if exported module is missing. 
**Actual behavior:**
Bundles without any errors

**Related Issues:**

**Code**
```ts
export * as n from "./a";

```

<details><summary><b>Output</b></summary>

```ts
System.register(["./a"], function (exports_1, context_1) {
    "use strict";
    var __moduleName = context_1 && context_1.id;
    return {
        setters: [
            function (n_1) {
                exports_1("n", n_1);
            }
        ],
        execute: function () {
        }
    };
});

```

</details>

<details><summary><b>Compiler Options</b></summary>

```json
{
  "compilerOptions": {
    "noImplicitAny": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "strictPropertyInitialization": true,
    "strictBindCallApply": true,
    "noImplicitThis": true,
    "noImplicitReturns": true,
    "useDefineForClassFields": false,
    "alwaysStrict": true,
    "allowUnreachableCode": false,
    "allowUnusedLabels": false,
    "downlevelIteration": false,
    "noEmitHelpers": false,
    "noLib": false,
    "noStrictGenericChecks": false,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "esModuleInterop": true,
    "preserveConstEnums": false,
    "removeComments": false,
    "skipLibCheck": false,
    "checkJs": false,
    "allowJs": false,
    "declaration": true,
    "experimentalDecorators": false,
    "emitDecoratorMetadata": false,
    "target": "ES2017",
    "module": "ESNext"
  }
}
```

</details>

**Playground Link:** [Provided](https://www.typescriptlang.org/play/?module=4&ssl=1&ssc=1&pln=2&pc=1#code/KYDwDg9gTgLgBAKjgQwM5wHZwGZQgWzgCIA6AemSIG4AoIA)

**Comment by kitsonk** [comment]
Your compiler options don't make sense for the issue you are describing.  There is no `outFile` and the module target is `ESNext`.  It will not produce a System bundle based on those options at all.

**Comment by grosto** [comment]
I opened the issue from playground and assumed it copied my compiler options from there. If you the open playground link you will see that module is set to System. 
```
  "compilerOptions": {
    "target": "es5",
    "module": "system",
    "outFile": "./bundle.js",
  }
```
Try these options. Anyways issue seems to be gone with TS v4.
